### ISSUE TITLE
TIFF with float sample format is not scRGB

### ISSUE THREAD

**Issue by DarthSim** [libvips/libvips#3621]
Hey there 👋

During my work on 16-bit float TIFFs support, I found out that vips loads TIFFs with float/double sample format as scRGB while they have sRGB colorspace despite having a non-integer sample format. Here's an example: [tiff7.tiff.zip](https://github.com/libvips/libvips/files/12422094/tiff7.tiff.zip)

If I do `vips colourspace tiff7.tiff tiff7.png srgb`, I'm getting the wrong colors.

Here's how it should look:
![tiff7_fixed](https://github.com/libvips/libvips/assets/380856/97a44b2e-4244-43d4-b525-358f7193e1aa)

And here's how it looks:
![tiff7_bugged](https://github.com/libvips/libvips/assets/380856/c1a28560-72a2-4add-9cf4-30093a832d26)

I was going to fit this myself, but I can't come out with the right approach:

1. Apply gamma correction during loading. Sounds legit, but we'll need to add gamma correction to the saving process too. But we should probably do so anyway.
2. Multiply pixel values to 65535 and load as RGB16. We'll need to handle this during saving and divide the pixel values back.
3. Just load them as RGB16. We'll lose precision, and this doesn't seem right in general.

What do you think?

**Comment by jcupitt** [comment]
Hi @DarthSim,

I wasn't sure if float TIFFs should have a gamma or not. Gamma is only really necessary for 8 bit data, so I thought probably not. And I think (though it was a long time ago) I found a program which could read and write float tiff which didn't use a gamma (radiance?).

Did you find something in the docs which clarifies this?

**Comment by jcupitt** [comment]
Ah your sample image has:

```
         <photoshop:ColorMode>3</photoshop:ColorMode>
         <photoshop:ICCProfile>sRGB IEC61966-2.1 (Linear RGB Profile)</photoshop:ICCProfile>
```

So it maybe should be linear?

I tried with vipsdisp and it displays it correctly:

![image](https://github.com/libvips/libvips/assets/580843/6bf33cbf-f6ed-4718-9513-92e4698040ee)

So I think it's maybe just poor scRGB support in the thing that converts image for save. I'll have a look.

**Comment by jcupitt** [comment]
nip2 displays it correctly as well, interestingly:

![image](https://github.com/libvips/libvips/assets/580843/34fc6a18-0255-4f17-85fc-cf7942fa1bb6)

**Comment by jcupitt** [comment]
Ah, this works:

```
$ vips colourspace tiff7.tiff x.png srgb
```

It's writing the correct PNG pixel values, but photoshop's linear RGB profile is still on there, so your image viewer makes it look washed out. 

If you do:

```
$ vips colourspace tiff7.tiff x.png[strip] srgb
```

You get:

![x](https://github.com/libvips/libvips/assets/580843/6c245705-677f-481b-8735-296ff22c6dd4)

So I suppose a solution might be to remove the ICC profile in any scRGB <-> sRGB conversion, since the profile can't work if the gamma changes. 

There's a related bug in the convert-for-save system:

```
$ VIPS_TRACE=1 vips copy tiff7.tiff x.png[strip]
vips cache : tiffload filename="tiff7.tiff" out=((VipsImage*) 0x555cf80891a0) flags=((VipsForeignFlags) VIPS_FOREIGN_SEQUENTIAL) access=((VipsAccess) VIPS_ACCESS_SEQUENTIAL) -
vips cache : copy in=((VipsImage*) 0x555cf80891a0) out=((VipsImage*) 0x555cf80983b0) -
vips cache : copy in=((VipsImage*) 0x555cf80983b0) out=((VipsImage*) 0x555cf809ddb0) -
vips cache+: cast in=((VipsImage*) 0x555cf80983b0) out=((VipsImage*) 0x555cf809cba0) format=((VipsBandFormat) VIPS_FORMAT_USHORT) -
vips cache : copy in=((VipsImage*) 0x555cf809cba0) out=((VipsImage*) 0x555cf80a1210) -
vips cache+: cast in=((VipsImage*) 0x555cf809cba0) out=((VipsImage*) 0x555cf809fe10) format=((VipsBandFormat) VIPS_FORMAT_UCHAR) -
vips cache+: linecache in=((VipsImage*) 0x7f52d007fc70) out=((VipsImage*) 0x7f52d0082f00) tile-height=16 access=((VipsAccess) VIPS_ACCESS_SEQUENTIAL) -
vips cache+: sequential in=((VipsImage*) 0x7f52d007fc70) out=((VipsImage*) 0x7f52d0081b10) tile-height=16 -
vips cache : pngsave_target in=((VipsImage*) 0x555cf80983b0) target=((VipsTarget*) 0x555cf809a370) strip=TRUE -
```

It's never calling scRGB -> sRGB, so the image comes out 256 times too dark.

**Comment by DarthSim** [comment]
I got it here: https://gitlab.gnome.org/GNOME/eog/-/issues/55. I removed the alpha channel because vips doesn't multiply it when converting from scRGB to sRGB and the image becomes almost fully transparent.

**Comment by DarthSim** [comment]
And again, most of the viewers show it correctly, not washed out.

**Comment by jcupitt** [comment]
I'm skeptical that the image is correct though. I think float 0-1 RGB values should not include a gamma, and whatever wrote that image is wrong.

Maybe we should ask for clarification on the libtiff mailing list?

**Comment by jcupitt** [comment]
Ah I found a post:

https://www.asmail.be/msg0054977954.html

So yes, photoshop uses gamma 1.0 for float RGB, and they control the tiff standard, so I think your image is wrong.

**Comment by jcupitt** [comment]
> The PhotometricInterpretation value of 2 (RGB data) specifies the NTSC gamma of 2.2 as a default

I don't think that applies to float RGB though. The tiff6 spec was written long before most people were working with float RGB. And sRGB isn't 2.2, of course.

> Maybe it's worth multiplying alpha during the load?

libvips uses unpremultiplied alpha, so I think it'd better to switch the alpha range for scRGB, it should be a small change I think.

**Comment by DarthSim** [comment]
Well, I spent $20 on renewing my Photoshop subscription just to find out that Photoshop doesn't even open this image complaining that the bitdepth is unsupported 😅 It opens the 16-bit float version of it though, and yeah, it shows it "washed out".

**Comment by jcupitt** [comment]
Oh haha sigh

Can you save as float32?

**Comment by DarthSim** [comment]
Yep. Photoshop opens 32-bit float TIFFs created with Photoshop. And it treats them as linear as well

**Comment by DarthSim** [comment]
No problem! I'm used to pushing my version until I see bulletproof arguments against it 😅 That was the most well-spent $20 today 😅
